lib/stack_master/validator.rb

```diff
@@ -10,8 +10,11 @@ def initialize(stack_definition, config)
     end
 
     def perform
+      parameter_hash = ParameterLoader.load(@stack_definition.parameter_files)
+      parameters = ParameterResolver.resolve(@config, @stack_definition, parameter_hash)
+
       StackMaster.stdout.print "#{@stack_definition.stack_name}: "
-      template_body = TemplateCompiler.compile(@config, @stack_definition.template_file_path, @stack_definition.compiler_options)
+      template_body = TemplateCompiler.compile(@config, @stack_definition.template_file_path, parameters, @stack_definition.compiler_options)
```

Comment:
Time to break this into multiple lines?

Reply:
Time to create a class for parameters me thinks. Wasn't going to make a structural change around existing code to much in this PR unless you really want me to

Reply:
Yeah lots of changes here already. Leave it as is